netaddr.go

```diff
@@ -859,14 +859,27 @@ func (ip *IP) UnmarshalBinary(b []byte) error {
 	return fmt.Errorf("netaddr: unexpected ip size: %v", len(b))
 }
 
-// IPPort is an IP & port number.
-//
-// It's meant to be used as a value type.
+// IPPort is an IP and a port number.
 type IPPort struct {
-	IP   IP
-	Port uint16
+	ip   IP
+	port uint16
 }
 
+// IPPortFrom returns an IPPort with ip ip and port port.
+func IPPortFrom(ip IP, port uint16) IPPort { return IPPort{ip: ip, port: port} }
+
+// WithIP returns an IPPort with ip ip and port p.Port().
+func (p IPPort) WithIP(ip IP) IPPort { return IPPort{ip: ip, port: p.port} }
+
+// WithIP returns an IPPort with ip p.IP() and port port.
```

Comment:
Worth documenting that these don't alloc? Or maybe just in package doc is sufficient if we add it there once?